lib/src/stores/node_list/node_list_store.dart

```diff
@@ -68,8 +68,8 @@ abstract class NodeListBase with Store {
     }
   }
 
-  static final nodeAddrRE = RegExp('^[0-9a-zA-Z.]+\$');
-  static final nodePortRE = RegExp('^[0-9]{1,5}\$');
+  static final nodeAddrRE = RegExp('^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\$|^[-0-9a-zA-Z.]{1,253}\$');
+  static final nodePortRE = RegExp('^([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$');
```

Comment:
> * The node regexp adds IP matching, but it's not actually accomplishing anything because anything that resembles an IP address is already accepted by the last part (which accepts up to 253 digits + dots).  In particular the intent here was to disallow an invalid IP (e.g. 123.456.789.101112) but that doesn't end up happening because such an invalid IP is accepted by the [0-9.]{1,253} alternative match at the end.

I'm glad that one of us is awake.
 
> I've added the hyphen match and the max-253 changes into #33.

Thanks. The allowing of the hyphen was the main thing I wanted to accomplish here.